addons/im_livechat/models/digest.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, api
+
+
+class Digest(models.Model):
+    _inherit = 'digest.digest'
+
+    kpi_im_livechat_rating = fields.Boolean('% of Happiness')
+    kpi_im_livechat_rating_value = fields.Float(digits=(16, 2), compute='_compute_kpi_livechat_rating_value')
+    kpi_im_livechat_conversations = fields.Boolean('Conversations handled')
+    kpi_im_livechat_conversations_value = fields.Integer(compute='_compute_kpi_livechat_conversations_value')
+    kpi_im_livechat_response = fields.Boolean('Time to answer(sec)', help="Time to answer the user in second.")
+    kpi_im_livechat_response_value = fields.Float(compute='_compute_kpi_livechat_response_value')
+
+    @api.multi
+    def _compute_kpi_livechat_rating_value(self):
+        for record in self:
+            start, end, company = record._get_kpi_compute_parameters()
+            channels = self.env['mail.channel'].search([('livechat_operator_id', '=', self.env.user.partner_id.id)])
```

Comment:
move this search outside of the loop for performance reason (as this is not depending on record)